### PR TITLE
Concent negotiation

### LINKED ISSUE #2752: Concent negotiation
Concent should be enabled in the TaskToCompute if and only if both parties have Concent service enabled.

If Provider has Concent enabled and Requestor has Concent disabled, then Requestor should not send TaskToCompute at all - there is no agreement, negotiation fails.

If Provider has Concent enabled and Requestor has Concent disabled and Requestor sent TaskToCompute with Concent distabled, they should reject any TTC that has concent disabled (CannotComputeTask).

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #2752

### CODE REVIEW

File: golem/task/tasksession.py [hunk 1]
```diff
@@ -524,6 +524,29 @@ def _react_to_task_to_compute(self, msg):
...
+                    reason=reason,
+                ),
+            )
+            self.task_computer.session_closed()
+            self.dropped()
+
+        reasons = message.CannotComputeTask.REASON
+
+        logger.debug("cnc enabled: %r", self.concent_service.enabled)
+        logger.debug("msg enabled: %r", msg.concent_enabled)
```
Comment (shadeofblue) [inline golem/task/tasksession.py]: maybe the debug message should say `msg concent enabled:` instead of just `msg enabled`?

Comment (shadeofblue) [reply golem/task/tasksession.py]: or even `ttc concent enabled:`

File: golem/task/tasksession.py [hunk 2]
```diff
@@ -524,6 +524,29 @@ def _react_to_task_to_compute(self, msg):
             self.task_computer.session_closed()
             self.dropped()
             return
+
+        def _cannot(reason):
```
Comment (shadeofblue) [inline golem/task/tasksession.py]: and I'd change it to either `_cannot_compute` or maybe `_cct` as the lone "cannot" sounds awkward-ish to me ...

**Review by shadeofblue — APPROVED** [review]

File: golem/task/tasksession.py (hunk 1 above)
Comment (jiivan) [reply golem/task/tasksession.py]: Those are just development artefacts. Thanks for pointing them out.

**Pull request merged**